tests/UKHO.ERPFacade.API.FunctionalTests/FunctionalTests/RoSWebhookScenarios.cs

```diff
@@ -0,0 +1,90 @@
+﻿using FluentAssertions;
+using NUnit.Framework;
+using UKHO.ERPFacade.API.FunctionalTests.Configuration;
+using UKHO.ERPFacade.API.FunctionalTests.Helpers;
+using UKHO.ERPFacade.API.FunctionalTests.Service;
+
+namespace UKHO.ERPFacade.API.FunctionalTests.FunctionalTests
+{
+    [TestFixture]
+    public class RoSWebhookScenarios
+    {
+        private RoSWebhookEndpoint _RosWebhookEndpoint { get; set; }
+        private readonly ADAuthTokenProvider _authToken = new();
+        private readonly string _projectDir = Path.GetFullPath(Path.Combine(Environment.CurrentDirectory));
+        //for local
+        //private readonly string _projectDir = Path.GetFullPath(Path.Combine(Environment.CurrentDirectory, "..\\..\\.."));
+        
+        [SetUp]
+        public void Setup()
+        {
+            _RosWebhookEndpoint = new RoSWebhookEndpoint();
+        }
+
+        [Test(Description = "WhenValidRoSEventInNewEncContentPublishedEventOptions_ThenWebhookReturns200OkResponse"), Order(0)]
+        public async Task WhenValidRoSEventInNewEncContentPublishedEventOptions_ThenWebhookReturns200OkResponse()
+        {
+            var response = await _RosWebhookEndpoint.OptionRosWebhookResponseAsync(await _authToken.GetAzureADToken(false));
+            response.StatusCode.Should().Be(System.Net.HttpStatusCode.OK);
+        }
+
+        [Test(Description = "WhenValidRoSEventInNewEncContentPublishedEventOptionsReceivedWithInvalidToken_ThenWebhookReturns401UnauthorizedResponse"), Order(1)]
```

Comment:
Please rename NewEncContentPublishedEventOptionsReceived to **RecordOfSalePublishedEventReceivedOptions**